torch/fx/passes/split_module.py

```diff
@@ -235,6 +315,23 @@ def instantiate_node_partition_mapping(node):
     if len(sorted_partitions) != len(partitions):
         raise RuntimeError("cycle exists between partitions!")
 
+    # Enter prelude
+    for regions_mapping in [autocast_regions, grad_regions]:
+        for node, regions in regions_mapping.items():
+            assert len(regions) > 0
+            partitions[str(regions[0])].environment[node] = node
+            for r in regions[1:]:
+                partition = partitions[str(r)]
+                new_node = partition.graph.create_node(
+                    op=node.op,
+                    target=node.target,
+                    args=tuple(arg for arg in node.args),
+                    kwargs={},
+                    type_expr=node.type,
+                )
+                new_node.meta = node.meta.copy()  # is it really a good idea to copy this?
```

Comment:
what does this typically contain?

what are the alternatives to copying it?
 - leaving it empty
 - adding a tainted copy, including some metadata that effectively says "i was created by split_module"
 - ...?

Reply:
Well meta is just a dict that can be used for anything. When the result is a Tensor, it often can store an example_value for instance.

Since we are dealing only with autocast/grad_enabled, I don't think we ought to be too worried, but let me inspect the metadata and see if there's anything that needs to be updated.

Reply:
There is prior art for copying the metadata when creating the new graph splits I.e. line 386, but my worry is more about the fact that we are here not copying 1-1 as per prior art but duplicating existing nodes multiple times across the subgraphs.

Reply:
Hmm, currently, the `meta` have the following values:

```
META PRELUDE {'creation_timestamp': 0}
META PRELUDE {'creation_timestamp': 0}
META PRELUDE {'creation_timestamp': 0}
META PRELUDE {'creation_timestamp': 0}
META EPILOGUE {'creation_timestamp': 0}
META EPILOGUE {'creation_timestamp': 0}
META EPILOGUE {'creation_timestamp': 0}
META EPILOGUE {'creation_timestamp': 0}
```

We may want to just update this for correctness sake - we can set it to be the creation timestamp of the first node in the partition.

Reply:
Actually creation_timestamp is only consumed within dynamo, so I think we don't need to care. (it's used for tracing checkpoint restoration)

Reply:
Created a PR to ensure this metadata lifetime is managed properly: https://github.com/pytorch/pytorch/pull/113517